### PR TITLE
pin to cmake <3.25 on CI to avoid bug with CUDA+conda during build

### PR DESCRIPTION
**Pull request by grlee77**
CMake 3.25 was released today and seems to have broken the CI builds. Trying to pin the version here to see if it resolves the issue.

attn: @jakirkham

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @robertmaynard (for awareness)

File: conda/recipes/cucim/meta.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ build:
 
 requirements:
   build:
-    - cmake >=3.18.0
+    - cmake >=3.23.1,<3.25
```
Comment (ajschmidt8) [inline conda/recipes/cucim/meta.yaml:23]:
```suggestion
    - cmake>=3.23.1,!=3.25.0
```

File: conda/recipes/libcucim/meta.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ build:
 
 requirements:
   build:
-    - cmake >=3.18.0
+    - cmake >=3.23.1,<3.25
```
Comment (ajschmidt8) [inline conda/recipes/libcucim/meta.yaml:22]:
```suggestion
    - cmake>=3.23.1,!=3.25.0
```

**Review by ajschmidt8 — COMMENTED** [review]
can we update these pinnings to match https://github.com/rapidsai/cudf/pull/12188/files?

That pinning is a little more forgiving than a max version pin.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by grlee77** [comment]
Thanks @ajschmidt8, I applied the suggestions and it should be ready to merge now

**Comment by grlee77** [comment]
@gpucibot merge

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**